src/main/java/com/getindata/connectors/http/internal/utils/ConfigUtils.java

```diff
@@ -0,0 +1,143 @@
+     *
+     * @param properties properties to extract keys from.
+     * @param keyPrefix prefix used to match property name with.
+     * @param valueClazz type to cast property values to.
+     * @return Map of propertyName to propertyValue.
+     */
+    public static <T> Map<String, T> propertiesToMap(
+            Properties properties,
+            String keyPrefix,
+            Class<T> valueClazz) {
+
+        Map<String, T> map = new HashMap<>();
+        for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+            if (entry.getKey() instanceof String) {
+                String key = (String) entry.getKey();
+                if (key.startsWith(keyPrefix)) {
+                    tryAddToConfigMap(properties, valueClazz, map, key);
+                }
+            } else {
+                throw new ConfigException(
+                    entry.getKey().toString(),
+                    entry.getValue(), "Key must be a string."
+                );
+            }
+        }
+        return map;
+    }
+
+    /**
+     * A utility method to extract last element from property name. This method assumes property to
+     * be in format as <b>{@code this.is.my.property.name}</b>, using "dot" as a delimiter. For this
+     * example the returned value would be <b>{@code name}</b>.
+     *
+     * @param propertyKay Property name to extract the last element from.
+     * @return property last element or the property name if {@code propertyKey} parameter had no
+     * dot delimiter.
+     * @throws ConfigException when invalid property such as null, empty, blank, ended with dot was
+     *                         used.
+     */
+    public static String extractPropertyLastElement(String propertyKay) {
+        if (StringUtils.isNullOrWhitespaceOnly(propertyKay)) {
+            throw new ConfigException("Provided a property name that is null, empty or blank.");
+        }
+
+        if (!propertyKay.contains(PROPERTY_NAME_DELIMITER)) {
+            return propertyKay;
+        }
+
+        int delimiterLastIndex = propertyKay.lastIndexOf(PROPERTY_NAME_DELIMITER);
+        if (delimiterLastIndex == propertyKay.length() - 1) {
+            throw new ConfigException(
+                String.format(
+                    "Invalid property - %s. Property name should not end with property delimiter.",
+                    propertyKay)
+            );
+        }
+
+        return propertyKay.substring(delimiterLastIndex + 1);
+    }
+
+    /**
+     * Flat map a given Map of header name and header value map to an array containing both header
+     * names and values. For example, header map of
+     * <pre>
+     *     Map.of(
+     *     header1, val1,
+     *     header2, val2
+     *     )
+     * </pre>
+     * will be converter to an array of:
+     * <pre>
+     *      String[] headers = {"header1", "val1", "header2", "val2"};
+     * </pre>
+     */
+    public static String[] flatMapToHeaderArray(Map<String, String> headerMap) {
```

Comment:
I'm not so sure about the name of that function, it is too low-level-expressive for my taste. I think just the `toHeaderArray` would be descriptive enough. Also, `flatMap` usually means conversion between the same containers.

Reply:
I have changed it to `toHeaderAndValueArray` I want to highlight in the name that we have not only header names but header name and header value, organized in pairs.

That's why I not used `toHeaderArray`